### PR TITLE
[RFR]deprecate and abstract configuration module usage

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
this moves our configuration object into a new package and reports all module style imports as deprecated

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit d5d695614d1261630ffc2da3eaa9b974755ee7cf:

`cfme/test_framework/config.py`:
- :grey_exclamation: Line 31:39: **S001** *found %s formatter*
- :grey_exclamation: Line 39:39: **S001** *found %s formatter*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit af9a09f5eb7543d1b5b1bae3e330fcc0c4d1332b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by mfalesni** [comment]
@RonnyPfannschmidt Can I get some code samples to play with?

File: cfme/test_framework/config.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import warnings
+import yaycl
+
+
+class Configuration(object):
+    def __init__(self):
+        self.yaycl_config = None
+
+    def configure(self, config_dir, crypt_key_file=None):
+        assert self.yaycl_config is None
```
Comment (mfalesni) [inline cfme/test_framework/config.py]: Is there any reason for blocking the reconfiguration here?

File: cfme/test_framework/config.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import os
+import warnings
+import yaycl
+
+
+class Configuration(object):
```
Comment (mfalesni) [inline cfme/test_framework/config.py]: Documentation comment please.

File: cfme/test_framework/config.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    def __init__(self):
+        self.yaycl_config = None
+
+    def configure(self, config_dir, crypt_key_file=None):
```
Comment (mfalesni) [inline cfme/test_framework/config.py]: Documentation comment please with explanation how this shall be used.

And if we can also pass just a string, without file (eg. as a parameter to py.test), that would be awesome.

File: cfme/test_framework/config.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        assert self.yaycl_config is None
+        if crypt_key_file and os.path.exists(crypt_key_file):
+            self.yaycl_config = yaycl.Config(
+                config_dir=config_dir,
+                crypt_key_file=crypt_key_file)
+        else:
+            self.yaycl_config = yaycl.Config(config_dir=config_dir)
+
+    def get_config(self, name):
```
Comment (mfalesni) [inline cfme/test_framework/config.py]: Docstring please.

**Comment by dajoRH** [comment]
Lint report for commit 4dd84f096154623968860680a247ecc38d9f5b25:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Pull request merged**